src/orion/algo/mofa/mofa.py

```diff
@@ -0,0 +1,410 @@
+        self.duplicates = defaultdict(list)
+        self.converged = False
+        self.n_trials = None
+
+        # Initialize a dictionary for the frozen values of each hyperparameter
+        self.frozen_param_values = {}
+
+        # Generate the initial set of trials
+        self.roi_space = self.space
+        self.current_trials_params = self._generate_trials_unfrozen_params(
+            self.roi_space
+        )
+
+    def _generate_trials_unfrozen_params(self, roi_space: Space) -> list[dict]:
+        olh_samples, self.cur_n_levels = sampler.generate_olh_samples(
+            roi_space, self.n_levels, self.strength, self.index, self.rng
+        )
+        self.n_trials = len(olh_samples)
+        olh_trials = sampler.generate_trials(olh_samples, roi_space)
+        return olh_trials
+
+    def seed_rng(self, seed: int) -> None:
+        """Seed the state of the random number generator.
+
+        Parameters
+        ----------
+        seed: int
+            Integer seed for the random number generator.
+
+        """
+        self.rng = np.random.RandomState(seed)
+
+    @property
+    def state_dict(self) -> dict:
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state_dict = super().state_dict
+        state_dict["rng_state"] = self.rng.get_state()
+        state_dict["completed_trials"] = copy.deepcopy(self.completed_trials)
+        state_dict["duplicates"] = copy.deepcopy(self.duplicates)
+        state_dict["converged"] = self.converged
+        state_dict["roi_space"] = self.roi_space
+        state_dict["frozen_param_values"] = copy.deepcopy(self.frozen_param_values)
+        if hasattr(self, "current_trials_params"):
+            state_dict["current_trials_params"] = copy.deepcopy(
+                self.current_trials_params
+            )
+        return state_dict
+
+    def set_state(self, state_dict: dict) -> None:
+        """Reset the state of the algorithm based on the given state_dict
+
+        Parameters
+        ----------
+        state_dict: dict
+            Dictionary representing state of an algorithm
+        """
+        super().set_state(state_dict)
+        self.seed_rng(0)
+        self.rng.set_state(state_dict["rng_state"])
+        self.completed_trials = copy.deepcopy(state_dict["completed_trials"])
+        self.duplicates = copy.deepcopy(state_dict["duplicates"])
+        self.converged = state_dict["converged"]
+        self.roi_space = copy.deepcopy(state_dict["roi_space"])
+        self.frozen_param_values = copy.deepcopy(state_dict["frozen_param_values"])
+        if "current_trials_params" in state_dict:
+            self.current_trials_params = copy.deepcopy(
+                state_dict["current_trials_params"]
+            )
+        elif hasattr(self, "current_trials_params"):
+            self.current_trials_params = []
+
+    def suggest(self, num: int) -> list[Trial]:
+        """Suggest a `num`ber of new sets of parameters.
```

Comment:
```suggestion
        """Suggest a number of new sets of parameters.
```
This also caused issues in other PR of plugin integration. It's hopefully the only issue with doc tests.